src/rdf_validator/endpoint.clj

```diff
@@ -12,8 +12,13 @@
     (let [r (repo/sail-repo)]
       (with-open [conn (repo/->connection r)]
         (log/info "Creating repository from directory: " (.getAbsolutePath f))
-        (doseq [df (.listFiles f)]
-          (pr/add conn (rdf/statements df))))
+        (loop [[file & files] (.listFiles f)]
+          (when file
+            (if (.isDirectory file)
+              (recur (concat files (.listFiles file)))
```

Comment:
Might we also not want to filter to only file extensions that are supported?  

```clojure
(defn supported-file? [f]
  (boolean (grafter-2.rdf4j.formats/filename->rdf-format f)))
```

Should hook into RDF4j's repository of RDF file extensions and parsers, and do the trick.

Reply:
Thinking on macos the filesystem is always littered with `.DS_Store` files; it drives me mad!

Reply:
Ah, I literally just deleted a pointless readme from ons-geo-graft and was wondering about doing this. I hadn't thought about those OS hidden files! I'll add this change, thanks!

Reply:
Resolved with 4b03ea2e6b756e728b13d97de720e47ca19bed55. Thanks for the snippet!